solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,530 @@
+     */
+    function burnIncentiveFee() external view returns (uint32, uint256) {
+        return (_burnIncentiveFee, _maxBurnIncentiveFeeAmount);
+    }
+
+    /**
+     * @dev allows the owner to set the burn incentive fee and its maximum amount
+     *
+     * @param newBurnIncentiveNetworkFee the percentage of the converted network tokens to be sent to the caller of the vortex (in units of PPM)
+     * @param newMaxBurnIncentiveFeeAmount the maximum incentive fee to be sent to the caller of the vortex
+     */
+    function setBurnIncentiveFee(uint32 newBurnIncentiveNetworkFee, uint256 newMaxBurnIncentiveFeeAmount)
+        external
+        ownerOnly
+        validFee(newBurnIncentiveNetworkFee)
+    {
+        emit BurnIncentiveFeeUpdated(
+            _burnIncentiveFee,
+            newBurnIncentiveNetworkFee,
+            _maxBurnIncentiveFeeAmount,
+            newMaxBurnIncentiveFeeAmount
+        );
+
+        _burnIncentiveFee = newBurnIncentiveNetworkFee;
+        _maxBurnIncentiveFeeAmount = newMaxBurnIncentiveFeeAmount;
+    }
+
+    /**
+     * @dev returns the total amount of the burned governance tokens
+     *
+     * @return total amount of the burned governance tokens
+     */
+    function totalBurnedAmount() external view returns (uint256) {
+        return _totalBurnedAmount;
+    }
+
+    /**
+     * @dev converts the provided tokens to governance tokens and burns them
+     *
+     * @param tokens the tokens to convert
+     */
+    function vortex(IERC20[] calldata tokens) external protected {
+        ITokenHolder feeWallet = networkFeeWallet();
+
+        // retrieve conversion strategy
+        Strategy memory strategy = vortexStrategy(tokens, feeWallet);
+
+        // withdraw all token/ETH amounts to the contract
+        feeWallet.withdrawTokensMultiple(tokens, address(this), strategy.amounts);
+
+        // convert all amounts to the network token and record conversion amounts
+        BancorNetwork network = bancorNetwork();
+
+        uint256[] memory networkTokenConversionAmounts = new uint256[](tokens.length);
+        uint256 grossNetworkTokenConversionAmount = 0;
+        uint256 totalGovTokenAmountToBurn = 0;
+
+        for (uint256 i = 0; i < strategy.paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = strategy.amounts[i];
+            if (amount == 0) {
+                continue;
+            }
+
+            address[] memory path = strategy.paths[i];
+            IERC20 token = IERC20(path[0]);
+            uint256 value = 0;
+
+            if (token == _govToken) {
+                // if the source token is the governance token, don't try to convert it either, but rather include it in
+                // the amount to burn
+                //
+                // Please note that networkTokenConversionAmounts[i] will remain 0 in this case
+                totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(amount);
```

Comment:
why do we need to track this? can't we just use the balanceOf(burner) before burning?

Reply:
Do you think it'd be cleaner to implicitly track it via balanceOf? I'm neutral.

Reply:
less stuff to pass around, less room for errors

Reply:
👍